app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardItemFragment.kt

```diff
@@ -0,0 +1,439 @@
+                            target.getPageTitle(forLanguageCode(targetLanguage!!)),
+                            target.displayTitle,
+                            target.description,
+                            target.thumbnailUrl,
+                            target.extractHtml
+                    )
+                    updateUI()
+                }, {
+                    L.e(it)
+                }))
+    }
+
+    private fun addCaption() {
+        callToActionButton.text = context?.getString(R.string.suggested_edits_feed_card_add_image_caption)
+        disposables.add(EditingSuggestionsProvider.getNextImageWithMissingCaption(langFromCode, MAX_RETRY_LIMIT)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .flatMap { title ->
+                    ServiceFactory.get(WikiSite(Service.COMMONS_URL)).getImageInfo(title, langFromCode)
+                            .subscribeOn(Schedulers.io())
+                            .observeOn(AndroidSchedulers.mainThread())
+                }
+                .subscribe({ response ->
+                    val page = response.query()!!.pages()!![0]
+                    if (page.imageInfo() != null) {
+                        val title = page.title()
+                        val imageInfo = page.imageInfo()!!
+
+                        sourceSummaryForEdit = PageSummaryForEdit(
+                                title,
+                                langFromCode,
+                                PageTitle(
+                                        Namespace.FILE.name,
+                                        StringUtil.removeNamespace(title),
+                                        null,
+                                        imageInfo.thumbUrl,
+                                        forLanguageCode(langFromCode)
+                                ),
+                                StringUtil.removeHTMLTags(title),
+                                imageInfo.metadata!!.imageDescription(),
+                                imageInfo.thumbUrl,
+                                null,
+                                imageInfo.timestamp,
+                                imageInfo.user,
+                                imageInfo.metadata
+                        )
+                        updateUI()
+                    }
+                }, {
+                    L.e(it)
+                }))
+    }
+
+    private fun translateCaption() {
+        cardActionType = TRANSLATE_CAPTION
+        callToActionButton.text = context?.getString(R.string.suggested_edits_feed_card_translate_image_caption,
+                app.language().getAppLanguageCanonicalName(targetLanguage))
+        if (targetLanguage!!.isEmpty()) {
+            return
+        }
+        var fileCaption: String? = null
+        disposables.add(EditingSuggestionsProvider.getNextImageWithMissingCaption(langFromCode, targetLanguage!!, MAX_RETRY_LIMIT)
```

Comment:
It took a pretty long time (and didn't finish) to load the card (`zh-hant`) and there's no error message or any other message shown on the card.

Reply:
Any updates?

Reply:
This is done. I also added an error view, so the progress doesn't keep going infinitely.